File: ts/256/155.ts
class MinStack {
  protected stack: number[] = []
  protected mins: number[] = [Infinity]

  constructor() {}

  push(val: number): void {
    this.stack.push(val)
    this.mins.push(Math.min(val, this.mins[this.mins.length - 1]))
  }

  pop(): void {
    this.stack.pop()
    this.mins.pop()
  }

  top(): number {
    return this.stack[this.stack.length - 1]
  }

  getMin(): number {
    return this.mins[this.mins.length - 1]
  }
}

File: py3/256/155.py
class MinStack:

    def __init__(self):
        """
        initialize your data structure here.
        """
        self._mins = []
        self._stack = []


    def push(self, x: int) -> None:
        self._stack.append(x)
        self._mins.append(x if not self._mins or self._mins[-1] > x else self._mins[-1])


    def pop(self) -> None:
        self._stack.pop()
        self._mins.pop()


    def top(self) -> int:
        return self._stack[-1]

    def getMin(self) -> int:
        return self._mins[-1]


# Your MinStack object will be instantiated and called as such:
# obj = MinStack()
# obj.push(x)
# obj.pop()
# param_3 = obj.top()
# param_4 = obj.min()